docs/providers/aws/guide/serverless.yml.md

```diff
@@ -457,6 +460,13 @@ layers:
 
 # The "Resources" your "Functions" use.  Raw AWS CloudFormation goes in here.
 resources:
+  Parameters:
+    # Parameters have all the attributes available from cloudformation parameter's capabilities
+    # To see more options: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html
+    ParameterName:
+      Type: String | Number | List | CommaDelimitedList | etc.
+      NoEcho: true | false # used to mask the parameter's value
```

Comment:
hmm.. I think it might be better to cover everything at `stackParameters` (also extending stack via `resources` as it's in v1 will probably be gone in v2, due to various implications see https://github.com/serverless/serverless/issues/6575)

Therefore I suggest to embrace following format:

```yaml
stackParameters:
  definitions:
    SomeParam:
      Type: ...
      NoEcho: ..
  data:
    - ParameterKey: SomeParam
      ParameterValue: Value
```

Reply:
That's an interesting idea, but there are issues with that approach.

1) The parameter's section in the "resources" section is already supported, its just not documented.
2) Issue #6575 has to do with overriding existing resources generated by the framework.  If in V2 you're planning to eliminate the resources section, there still needs to be a place for developers to put cloudformation components required by their solutions.  The parameters section at that point becomes important, but that should be considered a design update for V2 and not V1.  V2 should have parameters as a root level component, as provider is not really a great place long term.
3) Your requirement would create more merge issues between resource sections and the provider section.  This would be a poor design decision as it would cause errors which would be difficult to understand unless you had a very good understanding of the serverless framework.
4) The requirement you laid out would not be a small change. This change was meant to be a small incremental improvement.

If you did want to include definitions in the future of stackParameters, might I suggest that I add the "data" element from your example, but leave out the "definitions" section for now.  This would allow you to add the "definitions" section later and still allow for this incremental change to become available.

Reply:
>  If in V2 you're planning to eliminate the resources section,

We do not plan that. Have you read #6575 carefully? 

Currently we apply an unconditional deep merge, and that raises a lot of issues and makes room for a lot unintended errors. This design is very problematic. #6575 is about providing same functionality, but in _safe_ way, so no errors are accidentally produced. 

Same way with `Parameters` section, it works currently as undocumented feature by a side effect. Above I propose to put it on a map as documented feature, also I believe it should no not be at `resources`, as that part should extend just `Resources` section of CloudFormation.

Reply:
Your resource section right now is the only place to extend the resources, parameters, conditions, metadata and output section (basically anything that isn't currently available in the serverless framework.  You'll still need all of that available as it is extremely useful to have in a lot of enterprise based scenarios.

I had read #6575.  That was only resource specific.  I also read your comments which appear to focus on resource section being limited, which would also impact the ability to handle a wide variety of scenarios which are necessary for most enterprises and tools which plan to be published to the aws serverless repository.

Did you read my proposed compromise?  Would you be okay with that approach and be willing to merge a request with those changes?

Reply:
> Would you be okay with that approach and be willing to merge a request with those changes?

I don't think we should document `resources.Parameters` approach, as it works just by side effect, and will most likely be removed with v2.

To reduce confusion there's a plan to keep `resources` just to extend template  `Resources` section in v2.

Still if there's a demand to handle other top level template properties, we also need to agree for some room for them.

Maybe we should introduce a top level property `templateExtensions`, where we would allow adding `parameters` `outputs` etc. in controlled validated way.

Reply:
That makes sense.  So how about I remove the documentation for the resources:Properties and make the stack properties:
provider:
   stackProperties:
       data:
         - ParameterName
           ParameterValue

This leave room for what you're starting to envision for the rest of the stack properties element, while allowing parameters to be passed in now.  Would you be willing accept the pull request with those changes?

Reply:
I've updated proposal at https://github.com/serverless/serverless/issues/6575 to cover all other CF sections (Let me know what you think).

Otherwise yes, let's go as you propose, so remove `resources.Parameters` documentation, but I think we may keep `stackParameters` as you've already implemented , so parameters data put directly under it (in #6575 I've proposed to keep definitions at `cloudFormationAdditions.parameters`)

Reply:
I've removed the resources.Parameters documentation.  It should be ready now.